fs-objects.go

```diff
@@ -143,7 +145,73 @@ func (fs fsObjects) GetObjectInfo(bucket, object string) (ObjectInfo, error) {
 
 // PutObject - create an object.
 func (fs fsObjects) PutObject(bucket string, object string, size int64, data io.Reader, metadata map[string]string) (string, error) {
-	return putObjectCommon(fs.storage, bucket, object, size, data, metadata)
+	// Verify if bucket is valid.
+	if !IsValidBucketName(bucket) {
+		return "", BucketNameInvalid{Bucket: bucket}
+	}
+	// Check whether the bucket exists.
+	if !isBucketExist(fs.storage, bucket) {
+		return "", BucketNotFound{Bucket: bucket}
+	}
+	if !IsValidObjectName(object) {
+		return "", ObjectNameInvalid{
+			Bucket: bucket,
+			Object: object,
+		}
+	}
+
+	fileWriter, err := fs.storage.CreateFile(bucket, object)
```

Comment:
> but the separation was not driven by this but more because we have to do this for XL:

But still much of the codebase is still same, even till io.Copy and Md5sum validation.  Don't see any good reason to keep duplicates.  If its not clean then this approach is fine.